addons/purchase/tests/test_purchase_downpayment.py

```diff
@@ -0,0 +1,686 @@
+        self.assertEqual(len(downpayment_line), 1, 'PO line downpayment should be created on PO')
+        self.assertEqual(downpayment_line.price_unit, self.purchase_order.amount_total / 2, 'downpayment should have the correct amount')
+
+        invoice = self.purchase_order.invoice_ids[0]
+        downpayment_aml = invoice.line_ids.filtered(lambda l: not (l.display_type == 'line_section' and l.name == "Down Payments"))[0]
+        self.assertEqual(downpayment_aml.price_total, self.purchase_order.amount_total / 2, 'downpayment should have the correct amount')
+        self.assertEqual(downpayment_aml.price_unit, self.purchase_order.amount_total / 2, 'downpayment should have the correct amount')
+        self.post_bills()
+        self.assertEqual(downpayment_line.price_unit, self.purchase_order.amount_total / 2, 'downpayment should have the correct amount')
+
+    def test_downpayment_invoice_and_partial_credit_note(self):
+        """This test check that the downpayment line amount on the purchase order remains consistent"""
+        self.purchase_order.button_confirm()
+
+        self.make_downpayment(advance_payment_method='fixed', fixed_amount=100)
+
+        # Ensure the downpayment line on the sale order is correctly set to 100
+        downpayment_line = self.purchase_order.order_line.filtered(lambda l: l.is_downpayment and not l.display_type)
+        self.assertEqual(downpayment_line.price_unit, 100)
+
+        # post the downpayment invoice and ensure the downpayment_line amount is still 100
+        self.post_bills()
+        self.assertEqual(downpayment_line.price_unit, 100)
+
+        # Create a credit note for a part of the downpayment invoice and post it
+        po_invoice = max(self.purchase_order.invoice_ids)
+        credit_note_wizard = self.env['account.move.reversal'].with_context(
+            {'active_ids': [po_invoice.id], 'active_id': po_invoice.id, 'active_model': 'account.move'}).create({
+            'reason': 'no reason',
+            'journal_id': po_invoice.journal_id.id,
+            'date': '2020-02-01',
+        })
+        reversal_action = credit_note_wizard.refund_moves()
+        reverse_move = self.env['account.move'].browse(reversal_action['res_id'])
+        with Form(reverse_move) as form_reverse:
+            with form_reverse.invoice_line_ids.edit(0) as line_form:
+                line_form.price_unit = 20.0
+        reverse_move.action_post()
+        self.assertEqual(downpayment_line.price_unit, 80,
+                         "The downpayment line amount should be equal to the sum of the invoice and credit note amount")
+
+    def test_multi_company_invoice(self):
+        """
+            Checks that the company of the invoices generated in a multi-company environment using the
+           'purchase.advance.payment.wizard' wizard fit with the company of the PO and not with the current company.
```

Comment:
`'purchase.advance.payment.wizard' wizard matches the company of the PO and not with the current company.`